File: player_scraper.R
library(stringr)
library(rvest)
library(readr)
scrapy <- function(name){
  url <- sprintf("https://www.tennisexplorer.com/list-players/?search-text-pl=%s&country=", name)  
  download.file(url, destfile = "scrapedpage.html", quiet=TRUE)
  content <- read_html("scrapedpage.html")
  
  saved <- content %>%
    html_nodes(".t-name")%>%
    html_nodes("a")%>%
    html_attr("href")
  
  player_page <- sprintf("https://www.tennisexplorer.com%s", saved[1])
  download.file(player_page, destfile = "scrapedpage.html", quiet=TRUE)
  content <- read_html("scrapedpage.html")
  
  player_data <- content %>%
    html_nodes(".date")%>%
    html_text()
  
  scraped_data <- list("", "", "", "", "", "", "", "")
  
  for(p_info in player_data){
    splitted <- str_split(p_info, ": ")
    if(splitted[[1]][1] == "Country"){
      scraped_data[1] = splitted[[1]][2]
    }
    if((splitted[[1]][1] == "Height / Weight") || (splitted[[1]][1] == "Height")){
      scraped_data[2] = parse_number(splitted[[1]][2])
    }
    if(splitted[[1]][1] == "Age"){
      tmp <- str_split(splitted[[1]][2], " ")
      index <- 0
      for(i in tmp[[1]]){
        scraped_data[3+index] <- parse_number(i)
        index <- index+1
      }
    }
    if(splitted[[1]][1] == "Sex"){
      scraped_data[7] <- splitted[[1]][2]
    }
    if(splitted[[1]][1] == "Plays"){
      scraped_data[8] <- splitted[[1]][2]
    }
  }
  return(scraped_data)
}
add <- function(val){
  return(val)
}

File: player_scraper.jl
using HTTP, Gumbo, Cascadia, Dictionaries

#name_to_scrapy = "Daniil Medvedev";

function scrapy(name)
 	# substitute space with %20
	name = replace(name, " " => "%20")
	url = "https://www.tennisexplorer.com/list-players/?search-text-pl=$name";
	html_page = HTTP.get(url);

	r_parsed = parsehtml(String(html_page.body))
	player_tag = eachmatch(sel"a", eachmatch(sel".t-name", r_parsed.root)[1])[1].attributes["href"];
	

	url = "https://www.tennisexplorer.com$player_tag";
	html_page = HTTP.get(url);
	r_parsed = parsehtml(String(html_page.body))
	result = eachmatch(sel".date", r_parsed.root);

	player_data = dictionary(["Country" => "", "Height" => "", "Age" => "", "Sex" => "", "Plays" => "", "YearOfBirth" => ""])

	for el in result
		dat = split(nodeText(el), ":")
		if !occursin("Current", dat[1]) & !occursin("Today", dat[1])
			if dat[1] == "Age"
				tmp = split(replace(dat[2], r"\(|\.|\)" => ""), " ")
				player_data["Age"] = tmp[2]
				player_data["YearOfBirth"] = tmp[3] * "/" * tmp[4] * "/" * tmp[5]
			elseif (dat[1] == "Height / Weight") || (dat[1] == "Height")
				player_data["Height"] = split(replace(dat[2], r"cm|kg|/" => ""), " ")[2]
			else 
				player_data[dat[1]] = dat[2]
			end
		end
	end
	return player_data
end